packages/api-kit/src/SafeApiKit.ts

```diff
@@ -715,6 +719,110 @@ class SafeApiKit {
       }
     })
   }
+
+  /**
+   * Get the SafeOperations that were sent from a particular address.
+   * @param safeAddress - The Safe address to retrieve SafeOperations for
+   * @throws "Safe address must not be empty"
+   * @throws "Invalid Ethereum address {safeAddress}"
+   * @returns The SafeOperations sent from the given Safe's address
+   */
+  async getSafeOperationsByAddress(safeAddress: string): Promise<GetSafeOperationListResponse> {
+    if (!safeAddress) {
+      throw new Error('Safe address must not be empty')
+    }
+    const { address } = this.#getEip3770Address(safeAddress)
+
+    return sendRequest({
+      url: `${this.#txServiceBaseUrl}/v1/safes/${address}/safe-operations/`,
+      method: HttpMethod.Get
+    })
+  }
+
+  /**
+   * Get a SafeOperation by its hash.
+   * @param safeOperationHash The SafeOperation hash
+   * @throws "SafeOperation hash must not be empty"
+   * @throws "Not found."
+   * @returns The SafeOperation
+   */
+  async getSafeOperation(safeOperationHash: string): Promise<SafeOperationResponse> {
+    if (!safeOperationHash) {
+      throw new Error('SafeOperation hash must not be empty')
+    }
+
+    return sendRequest({
+      url: `${this.#txServiceBaseUrl}/v1/safe-operations/${safeOperationHash}/`,
+      method: HttpMethod.Get
+    })
+  }
+
+  /**
+   * Create a new 4337 SafeOperation for a Safe.
+   * @param addSafeOperationProps - The configuration of the SafeOperation
+   * @throws "Safe address must not be empty"
+   * @throws "Invalid Safe address {safeAddress}"
+   * @throws "Module address must not be empty"
+   * @throws "Invalid module address {moduleAddress}"
+   * @throws "SafeOperation is not signed by the given signer {signerAddress}"
+   */
+  async addSafeOperation({
+    moduleAddress: moduleAddressProp,
+    safeAddress: safeAddressProp,
+    safeOperation,
+    signer
+  }: AddSafeOperationProps): Promise<void> {
+    let safeAddress: string, moduleAddress: string
+
+    if (!safeAddressProp) {
+      throw new Error('Safe address must not be empty')
+    }
+    try {
+      safeAddress = this.#getEip3770Address(safeAddressProp).address
+    } catch (err) {
+      throw new Error(`Invalid Safe address ${safeAddressProp}`)
+    }
+
+    if (!moduleAddressProp) {
+      throw new Error('Module address must not be empty')
+    }
+
+    try {
+      moduleAddress = this.#getEip3770Address(moduleAddressProp).address
+    } catch (err) {
+      throw new Error(`Invalid module address ${moduleAddressProp}`)
+    }
+
+    const signerAddress = await signer.getAddress()
+    const signature = safeOperation.getSignature(signerAddress)
```

Comment:
I was looking at the server-side code and this field seems to be validated like this:
```
safe_owners = self._get_owners(safe_address)
parsed_signatures = SafeSignature.parse_signature(
    signature, safe_operation_hash, safe_operation_hash_preimage
)
owners_processed = set()
safe_signatures = []
for safe_signature in parsed_signatures:
    # do stuff
```
Please just check if those are not every owner's concatenated signature. I think the `SafeOperation#encodedSignatures()` or something.